### DISCUSSION TITLE
[Prettier Plugin] Get the plugin working with grouping

### DISCUSSION THREAD

**Discussion by amritk** [tailwindlabs/tailwindcss#12398] [Ideas]
Hello, @rschristian has created a [wonderful plugin](https://github.com/rschristian/tailwind-grouping/issues/1) which solves the [grouping issue](https://github.com/tailwindlabs/tailwindcss/discussions/11701) with no downsides. Would it be possible to get the prettier plugin to work with it?

Best case sort the groups, worst case don't do anything? Currently it boots all classes but two out of the group. I would be willing to work on it if this is something that would be accepted. 

currently it changes `hover:(bg-red gap-2 mt-9 p-64)` -> `hover:(bg-red p-64) mt-9 gap-2`

**Comment by amritk** [comment]
Alright I got the plugin working with a small tweak to the regex. Now the question is, would it be worth merging this new regex? Or exposing the class splitting regex from sorting.js as an option so people can use their own regex? In the meanwhile I will just fork this plugin so vite users can start using grouping!

**Comment by u3u** [comment]
Thank you! According to your PR https://github.com/tailwindlabs/prettier-plugin-tailwindcss/pull/227/files, I made modifications using `pnpm patch`, which can solve my problem: https://github.com/tailwindlabs/tailwindcss/discussions/12456

**Reply by amritk** [reply]
Unfortunately I didn't solve the sorting in the pr

**Reply by u3u** [reply]
It's okay! As long as the grouping and original style are not destroyed, I can accept it!
